catalogue/webapp/components/Sort/Sort.tsx

```diff
@@ -42,22 +42,25 @@ const Sort: FunctionComponent<Props> = ({
   defaultValues,
 }) => {
   const router = useRouter();
+  const { sortOrder: currentSortOrder, sort: currentSortType } = router.query;
 
   const [isComponentMounted, setIsComponentMounted] = useState(false);
   useEffect(() => setIsComponentMounted(true), []);
 
-  const [sortOrder, setSortOrder] = useState(router?.query?.sortOrder);
-  const [sortType, setSortType] = useState(router?.query?.sort);
+  const [sortOrder, setSortOrder] = useState(currentSortOrder);
+  const [sortType, setSortType] = useState(currentSortType);
 
   useEffect(() => {
-    const queryParams = { ...router.query, sortOrder, sort: sortType };
-    const newQuery = propsToQuery(queryParams);
+    // Only push changes if the sort order is a new one than currently used
+    if (sortOrder !== currentSortOrder && sortType !== currentSortType) {
```

Comment:
Does this condition need to be `||` instead of `&&`?

e.g. if I run this locally and go to http://localhost:3000/search/works?sort=production.dates&sortOrder=desc, I don't see any changes if I toggle between "Newest to oldest" or "Oldest to newest", because the sortType is staying the same each time.

Reply:
This is an excellent catch, thanks. (and thanks for testing it locally)

Reply:
Behaviour should be fixed now - as mentioned in standup I'll look at adding a test, but if you could check if the bug is fixed maybe I'd do that separately?